vue/src/components/loki/LogLabelSelect.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <v-chip
+    v-model="isLabelSelected"
+    label
+    small
+    class="ma-1"
+    v-if="label"
+    :color="isLabelSelected ? 'blue' : 'grey lighten-4'"
+    :class="{ active: isLabelSelected }"
+    @click="setIsLabelSelected"
+  >
+    {{ label.name }}
+  </v-chip>
+</template>
+
+<script lang="ts">
+import { defineComponent, shallowRef, PropType, watch, computed } from '@vue/composition-api'
+
+// Composables
+import { UseDateRange } from '@/use/date-range'
+import { Label } from '@/components/loki/logql'
+
+export default defineComponent({
+  name: 'LogLabelSelect',
+  props: {
+    dateRange: {
+      type: Object as PropType<UseDateRange>,
+      required: true,
+    },
+    label: {
+      type: Object as PropType<Label>,
+      require: true,
+    },
+  },
+  setup(props, ctx) {
+    const isLabelSelected = shallowRef(props.label?.selected)
+    const setLabelSelected = computed({
+      get: () => isLabelSelected.value,
+      set: (value) => {
+        isLabelSelected.value = value
+      },
+    })
+    watch(
+      () => props.label?.selected,
+      (label) => {
+        isLabelSelected.value = label || false
+      },
+      { immediate: true },
+    )
```

Comment:
Looks like the issue you've tried to explain in the chat. Let's discuss this together later.